modules/core/include/opencv2/core/types.hpp

```diff
@@ -60,8 +60,79 @@
+{                                                                                                  \
+    _this = _this & val;                                                                           \
+    return _this;                                                                                  \
+}                                                                                                  \
+
+#define CV_FLAGS_ENUM_XOR_EQ(EnumType, Arg1Type)                                                   \
+static inline EnumType& operator^=(EnumType& _this, const Arg1Type& val)                           \
+{                                                                                                  \
+    _this = _this ^ val;                                                                           \
+    return _this;                                                                                  \
+}                                                                                                  \
+
+#define CV_FLAGS_ENUM(EnumType)                                                                    \
+CV_FLAGS_ENUM_NOT   (EnumType);                                                                    \
+CV_FLAGS_ENUM_OR    (EnumType, EnumType, EnumType);                                                \
+CV_FLAGS_ENUM_AND   (EnumType, EnumType, EnumType);                                                \
+CV_FLAGS_ENUM_XOR   (EnumType, EnumType, EnumType);                                                \
+                                                                                                   \
+CV_FLAGS_ENUM_OR    (EnumType, int,      EnumType);                                                \
+CV_FLAGS_ENUM_AND   (EnumType, int,      EnumType);                                                \
+CV_FLAGS_ENUM_XOR   (EnumType, int,      EnumType);                                                \
+CV_FLAGS_ENUM_OR    (EnumType, EnumType, int);                                                     \
+CV_FLAGS_ENUM_AND   (EnumType, EnumType, int);                                                     \
+CV_FLAGS_ENUM_XOR   (EnumType, EnumType, int);                                                     \
+                                                                                                   \
+CV_FLAGS_ENUM_OR_EQ (EnumType, EnumType);                                                          \
+CV_FLAGS_ENUM_AND_EQ(EnumType, EnumType);                                                          \
+CV_FLAGS_ENUM_XOR_EQ(EnumType, EnumType);                                                          \
+CV_FLAGS_ENUM_OR_EQ (EnumType, int);                                                               \
+CV_FLAGS_ENUM_AND_EQ(EnumType, int);                                                               \
+CV_FLAGS_ENUM_XOR_EQ(EnumType, int);                                                               \
 
```

Comment:
@alalek @hrnr I am glad there is no limit for macros length ;)

Reply:
Nice work. Keep going!

I think that int is enough for the overloads as enums convert implicitly to ints (we have only int enums). Moreover, I think we should get rid of those overloads one day to disallow `AccessFlags flags = cv::RANSAC | ACCESS_WRITE;`. Casts would need to be added to some places though.

Reply:
The good news is, if `cv::RANSAC` is not an `int`, `AccessFlags flags = cv::RANSAC | ACCESS_WRITE` would not work.
That is another reason to avoid using `int`, and using enum types instead.